tools/version-tracker/pkg/ecrpublic/ecrpublic.go

```diff
@@ -0,0 +1,58 @@
+package ecrpublic
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	//"regexp"
+	"encoding/json"
+
+	"github.com/aws/eks-anywhere/pkg/semver"
+
+	"github.com/aws/eks-anywhere-build-tooling/tools/version-tracker/pkg/util/command"
+)
+
+func GetLatestRevision(imageRepository, currentRevision string) (string, bool, error) {
+	var latestRevision string
+	currentRevisionSemver, err := semver.New(currentRevision)
+	if err != nil {
+		return "", false, fmt.Errorf("getting semver for current version: %v", err)
+	}
+
+	skopeoListTagsCmd := exec.Command("skopeo", "list-tags", fmt.Sprintf("docker://%s", imageRepository))
+	listTagsOutput, err := command.ExecCommand(skopeoListTagsCmd)
+	if err != nil {
+		return "", false, fmt.Errorf("running Go version command: %v", err)
+	}
+
+	var tagsList interface{}
+	err = json.Unmarshal([]byte(listTagsOutput), &tagsList)
+	if err != nil {
+		return "", false, fmt.Errorf("unmarshaling output of Skopeo list-tags command: %v", err)
+	}
+
+	ciliumTags := tagsList.(map[string]interface{})["Tags"].([]interface{})
+
+	latestRevisionSemver := currentRevisionSemver
```

Comment:
something for later. we may need to handle multiple minor versions, say like in a release branch where we are using 1.13 but in main we are using 1.14. this situation would apply to most projects really, not just cilium/ecr based.

Reply:
Yeah agreed. For now, this automation only upgrades projects on main branch and we backport to release branch on a case-by-case basis.